Framework/CurveFitting/src/Algorithms/LeBailFit.cpp

```diff
@@ -632,29 +636,29 @@ void LeBailFit::execRefineBackground() {
   m_backgroundFunction->function(domain, values);
   for (size_t i = 0; i < numpts; ++i) {
     backgroundvalues[i] = values[i];
-    m_outputWS->dataY(INPUTPUREPEAKINDEX)[i] =
-        m_dataWS->readY(m_wsIndex)[i] - values[i];
+    m_outputWS->mutableY(INPUTPUREPEAKINDEX)[i] =
+        m_dataWS->y(m_wsIndex)[i] - values[i];
   }
   parammap = convertToDoubleMap(m_funcParameters);
   m_lebailFunction->setProfileParameterValues(parammap);
-  calculateDiffractionPattern(m_outputWS->readX(INPUTPUREPEAKINDEX),
-                              m_outputWS->readY(INPUTPUREPEAKINDEX), false,
-                              true, backgroundvalues, valueVec, outputR);
+  calculateDiffractionPattern(m_outputWS->x(INPUTPUREPEAKINDEX).rawData(),
+                              m_outputWS->y(INPUTPUREPEAKINDEX).rawData(),
+                              false, true, backgroundvalues, valueVec, outputR);
 
   g_log.notice() << "[RefineBackground] Best Rwp = " << bestR.Rwp
                  << ",  vs. recovered best Rwp = " << outputR.Rwp << ".\n";
 
   // 4. Add data (0: experimental, 1: calcualted, 2: difference)
   for (size_t i = 0; i < numpts; ++i) {
-    m_outputWS->dataY(1)[i] = valueVec[i] + backgroundvalues[i];
-    m_outputWS->dataY(2)[i] = vecY[i] - (valueVec[i] + backgroundvalues[i]);
+    m_outputWS->mutableY(1)[i] = valueVec[i] + backgroundvalues[i];
+    m_outputWS->mutableY(2)[i] = vecY[i] - (valueVec[i] + backgroundvalues[i]);
```

Comment:
In general, calling `dataY` or `mutableY` in a loop over bins should be avoided, since it has non-negligible cost due to the copy-on-write mechanism. Here it is probably fine since it is done only for 2 histograms, but if you have a big workspace with many histograms you can see the difference.